Foundation/NSPathUtilities.swift

```diff
@@ -345,7 +345,100 @@ public extension NSString {
     /// - Experiment: This is a draft API currently under consideration for official import into Foundation
     /// - Note: Since this API is under consideration it may be either removed or revised in the near future
     public func completePathIntoString(inout outputName: NSString?, caseSensitive flag: Bool, inout matchesIntoArray outputArray: [NSString], filterTypes: [String]?) -> Int {
```

Comment:
I believe `NSURL(string: _storage)` really should be `NSURL(fileURLWithPath: _storage)` so it could be standardized, but core method of `NSURL` that does this is not implemented yet. There is `FIXME` comment but it is not very clear it seems. For now only paths without any symlinks etc supported.